File: DSA/DSA.DataStructures/Linear/Stack/LLStack/Stack.cs
using DSA.DataStructures.Linear.LinkedLists.DoubleLinkedListNS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSA.DataStructures.Linear.Stack.LLStack
{
    public class Stack : IStack
    {
        private readonly DoubleLinkedList linkedList = new();
        public void Clear()
        {
            linkedList.Clear();
        }

        public bool IsEmpty()
        {
            return linkedList.GetLegnth() == 0;
        }

        public int Peek()
        {
            return linkedList.PeekLast();
        }

        public int Pop()
        {
            return linkedList.RemoveLast();
        }

        public void Push(int item)
        {
            linkedList.AddLast(item);
        }
    }
}

File: study/src/internal/datastructures/stack/Stack.go
package stack

import "learningDSA/src/internal/datastructures/linkedlists"

type Stack struct {
	l linkedlists.SingleLinkedList
}

func (s *Stack) Push(d int) {
	s.l.AddFirst(d)
}

func (s *Stack) Pop() (int, bool) {
	return s.l.RemoveFirst()
}

func (s *Stack) Peek() (int, bool) {
	return s.l.PeekFirst()
}

func (s *Stack) String() string {
	return s.l.String()
}

func (s *Stack) Length() int {
	return s.l.Length
}